### PR TITLE
Added more errors!

### PR DESCRIPTION
**Pull request by Shariefie**
Check the errors

### CODE REVIEW

File: Core/Inc/API.h [hunk 1]
```diff
@@ -24,19 +24,27 @@
...
 #define LETTER_BITS				31	//Letter bits
 #define LETTER_WIDTH_BITS		32	//Character bits
 #define FONT_SIZE_POSITION		3	//Position of font size in the array for each fonts.
-#define RADIUS_INCREMENT_CIRCLE 3600
+#define COLORMAX				255	//Color between 0 and 255
+#define COLORMIN				0	//Min color digit.
+#define LINE_WIDTH				10	//								<----*
+#define X_BEGIN					0
+#define Y_BEGIN					0
+#define MAX_FIGURE_POINTS		10
```
Comment (bestehustudentEU) [inline Core/Inc/API.h]: this define is not needed.

File: Core/Src/API.c [hunk 2]
```diff
@@ -218,19 +263,26 @@ void drawRect(uint16_t x_pos, uint16_t y_pos, uint16_t length, uint16_t width, u
...
 		plaats_y = (VGA_DISPLAY_Y/2) + round(radius * sin(i));  // Y = r*sine(θ)
-#ifdef DEBUG_CIRCLE_PLAATS
+		if ((plaats_x < 0 || plaats_x > VGA_DISPLAY_X) || (plaats_y < 0 || plaats_y > VGA_DISPLAY_Y))
+		{
+			softonErrorHandler(ERROR_CIRCLE_OUT_OF_RANGE);
+			return;
+		}
+		#ifdef DEBUG_CIRCLE_PLAATS
 		if(i=M_PI) printf("plaats x is \t %d \n plaats y is \t %d",plaats_x, plaats_y);
 #endif
```
Comment (Tjerk258) [inline Core/Src/API.c:287]: geen ifdef wel endif?

File: Core/Src/main.c [hunk 3]
```diff
@@ -112,8 +112,17 @@ int main(void)
...
+  //drawText(0, 20, VGA_COL_BLUE, "ABCDEFGHIJKL", "Arial", 1, 1);
+  //drawBitmap(0,0,0);
+  //drawBitmap(0,50,50);
+  //drawText(3, 2, VGA_COL_BLUE, "ABCDEFGHIJKL", "Consolas", 1, 1);
+  //drawText(0, 20, VGA_COL_BLUE, "ABCDEFGHIJKL", "Arial", 1, 1);
+  //drawLines(1,1,50,50,254,11);
+  //drawRect(10,10,5,352,VGA_COL_BLUE,1);
+  drawCircle(120,160,500,VGA_COL_BLUE);
+  //drawText(0, 20, VGA_COL_BLUE, "ABCDEFGHIJKL", "Arial", 1, 1);
+  //drawText(0, 20, VGA_COL_BLUE, "ABCDEFGHIJKL", "Arials", 1, 1);
```
Comment (Tjerk258) [inline Core/Src/main.c]: Onnodige comments

File: Core/Inc/error.h [hunk 4]
```diff
@@ -7,13 +7,24 @@
...
 #include <stdio.h>
 
-#define ERROR_BITMAP_NOT_FOUND			1
-#define ERROR_FIGURE_OUT_OFF_RANGE		2
-#define ERROR_FONT_NOT_FOUND			3
-#define ERROR_FONT_OUT_OF_RANGE			4
-#define ERROR_FONTSIZE_OUT_OF_RANGE 	5
-#define ERROR_FONTSTYLE_OUT_OF_RANGE 	6
-#define ERROR_BITMAP_OUT_OF_RANGE		7
+#define ERROR_BITMAP_NOT_FOUND				1
```
Comment (bestehustudentEU) [inline Core/Inc/error.h:10]: defines has diference spacing.

**Review by Tjerk258 — CHANGES_REQUESTED** [review]
ziet er goed uit alleen error functie moet in logic_layer aangeroepen worden

File: Core/Src/API.c [hunk 5]
```diff
@@ -26,6 +26,16 @@
 
 void myLijntekenaar(uint16_t x_begin, uint16_t y_begin, uint16_t x_eind, uint16_t y_eind,uint8_t kleur)
 {
+	if((x_begin < 0) || (y_begin < 0) || (x_eind > VGA_DISPLAY_X) || (y_eind < VGA_DISPLAY_Y))
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: not 100%, x_begin,y_begin can be higher than display and x_eind ,y_eind can be lower than zero

File: Core/Src/API.c [hunk 6]
```diff
@@ -182,12 +191,32 @@ void myLijntekenaar(uint16_t x_begin, uint16_t y_begin, uint16_t x_eind, uint16_
 
 void drawLines(uint16_t x_begin, uint16_t y_begin, uint16_t x_eind, uint16_t y_eind, uint8_t kleur, uint8_t lijn_dikte)
 {
+	if(x_begin < 0x00)
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: define forgotten.

File: Core/Src/API.c [hunk 7]
```diff
@@ -182,12 +191,32 @@ void myLijntekenaar(uint16_t x_begin, uint16_t y_begin, uint16_t x_eind, uint16_
...
+	{
+		softonErrorHandler(ERROR_LINE_OUT_OF_RANGE);
+		return;
+	}
+	if(((x_begin+x_eind) > VGA_DISPLAY_X) || ((y_begin+y_eind) > VGA_DISPLAY_Y))
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: ? do the same as my tekenaar.

File: Core/Src/API.c [hunk 8]
```diff
@@ -182,12 +191,32 @@ void myLijntekenaar(uint16_t x_begin, uint16_t y_begin, uint16_t x_eind, uint16_
...
+	{
+		softonErrorHandler(ERROR_LINE_LENGTH_OUT_OF_RANGE);
+		return;
+	}
+	if (kleur > COLORMAX || kleur < COLORMIN)
+	{
+		softonErrorHandler(ERROR_LINE_COLOR_OUT_OF_RANGE);
+		return;
+	}
+	if (lijn_dikte > LINE_WIDTH)
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: smaller than, not needed?

File: Core/Src/API.c [hunk 9]
```diff
@@ -202,9 +231,25 @@ void drawLines(uint16_t x_begin, uint16_t y_begin, uint16_t x_eind, uint16_t y_e
 
 void drawRect(uint16_t x_pos, uint16_t y_pos, uint16_t length, uint16_t width, uint8_t kleur, uint8_t filled)
 {
+	if(((x_pos+length) > VGA_DISPLAY_X) || ((y_pos+width) > VGA_DISPLAY_Y) || ((x_pos+length) < VGA_DISPLAY_X) || ((y_pos+width) < VGA_DISPLAY_Y))
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: compleet fout!
(x_pos+length) groter of lager dan vga display?

File: Core/Src/API.c [hunk 10]
```diff
@@ -202,9 +231,25 @@ void drawLines(uint16_t x_begin, uint16_t y_begin, uint16_t x_eind, uint16_t y_e
...
+	}
+	if (filled > 1)
+	{
+		softonErrorHandler(ERROR_FILLED_OUT_OF_RANGE);
+		return;
+	}
 	uint8_t i = 0;
 	if (filled == 0) // If the rectangle does not need to be filled
 	{
+
```
Comment (bestehustudentEU) [inline Core/Src/API.c:248]: why did you add spacing?

File: Core/Src/API.c [hunk 11]
```diff
@@ -218,19 +263,26 @@ void drawRect(uint16_t x_pos, uint16_t y_pos, uint16_t length, uint16_t width, u
 	else
 	{
 		for (i = y_pos; i < width +y_pos; i++) // Loops around the y axis
+		{
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: why did you add curly brackets?

File: Core/Src/API.c [hunk 12]
```diff
@@ -218,19 +263,26 @@ void drawRect(uint16_t x_pos, uint16_t y_pos, uint16_t length, uint16_t width, u
...
 			myLijntekenaar(x_pos, i, x_pos + length-1, i, kleur); // Draws around the x axis
+		}
 	}
 }
 
-void drawCircle(uint16_t x_pos, uint16_t y_pos, uint8_t radius, uint8_t kleur)
+	void drawCircle(uint16_t x_pos, uint16_t y_pos, uint8_t radius, uint8_t kleur)
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: spacing?

File: Core/Src/main.c [hunk 13]
```diff
@@ -112,8 +112,17 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   //drawBitmap(0,0,0);
   //drawBitmap(0,50,50);
-  drawText(0, 0, VGA_COL_BLUE, "ABCDEFGHIJKL", "Consolas", 0, 0);
-  drawText(0, 20, VGA_COL_BLUE, "ABCDEFGHIJKL", "Arial", 1, 1);
+  //drawText(0, 0, VGA_COL_BLUE, "ABCDEFGHIJKL", "Consolas", 0, 0);
```
Comment (bestehustudentEU) [inline Core/Src/main.c]: delete comment

**Review by bestehustudentEU — COMMENTED** [review]
see comments

File: Core/Src/API.c (hunk 2 above)
Comment (Shariefie) [inline Core/Src/API.c]: Solved!

File: Core/Src/main.c (hunk 3 above)
Comment (Shariefie) [reply Core/Src/main.c]: Deleted

File: Core/Inc/API.h (hunk 1 above)
Comment (Shariefie) [reply Core/Inc/API.h]: Solved!

File: Core/Inc/error.h (hunk 4 above)
Comment (Shariefie) [reply Core/Inc/error.h:10]: Solved!

File: Core/Src/API.c (hunk 6 above)
Comment (Shariefie) [reply Core/Src/API.c]: This is your own code written!!

Comment (Shariefie) [reply Core/Src/API.c]: But i solved it!

File: Core/Src/API.c (hunk 7 above)
Comment (Shariefie) [reply Core/Src/API.c]: Solved!

File: Core/Src/API.c (hunk 9 above)
Comment (Shariefie) [reply Core/Src/API.c]: Solved!

File: Core/Src/API.c (hunk 10 above)
Comment (Shariefie) [reply Core/Src/API.c:248]: I did not add space, github issue i guess!

File: Core/Src/API.c (hunk 11 above)
Comment (Shariefie) [reply Core/Src/API.c]: Nothing found, github issue.

File: Core/Src/API.c (hunk 12 above)
Comment (Shariefie) [reply Core/Src/API.c]: Nothing found.....

File: Core/Src/main.c (hunk 13 above)
Comment (Shariefie) [reply Core/Src/main.c]: deleted already..

**Review by Tjerk258 — APPROVED** [review]

**Review by bestehustudentEU — APPROVED** [review]

**Pull request merged**